pennylane/qchem/convert_openfermion.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed to convert OpenFermion ``QubitOperator`` objects to PennyLane ``Sum`` and ``LinearCombination`` and vice versa.
+"""
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.qchem.convert import _openfermion_to_pennylane, _pennylane_to_openfermion
+
+
+def from_openfermion(of_qubit_operator, wires=None, tol=None):
+    r"""Convert OpenFermion ``QubitOperator`` to a ``LinearCombination`` object in PennyLane representing a linear combination of qubit operators.
+
+    Args:
+        qubit_operator (QubitOperator): fermionic-to-qubit transformed operator in terms of
+            Pauli matrices
+        wires (Wires, list, tuple, dict): Custom wire mapping used to convert the qubit operator
+            to an observable terms measurable in a PennyLane ansatz.
+            For types Wires/list/tuple, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted.
+            If None, will use identity map (e.g. 0->0, 1->1, ...).
```

Comment:
I'm curious if we need a wire mapping here if we already have [map_wires](https://docs.pennylane.ai/en/stable/code/api/pennylane.map_wires.html)?

Reply:
@trbromley Valid point, thank you! We can indeed just use `map_wires` after we have converted a `QubitOperator` from OpenFermion to a `LinearCombination` in PennyLane. I have two options: 
- Defer the functionality of the wire mapping from `from_openfermion` to `map_wires`? 
- Or adjust the implementation in the [`_openfermion_to_pennylane`](https://github.com/PennyLaneAI/pennylane/blob/c14ce7bc18b7c38e5246f2a5451ffb3cd0d10077/pennylane/qchem/convert.py#L121) method to use the `map_wires` functionality to avoid code duplication?

The `wires` argument is still needed in `to_openfermion`, as the function should work for the objects`Sum`,`LinearCombination`, `FermiWord` and `FermiSentence` and the `map_wires` is currently not supported for `FermiWord` and `FermiSentence`.

Reply:
Thanks! I'd favour removing the `wires` argument in `from_openfermion` for now, and we can always add it in if someone complains :smile:

> The wires argument is still needed in [to_openfermion](https://github.com/PennyLaneAI/pennylane/blob/ead94bd035dc1e0fae494e41b8128a34647684b0/pennylane/qchem/convert_openfermion.py#L73), as the function should work for the objectsSum,LinearCombination, FermiWord and FermiSentence and the map_wires is currently not supported for FermiWord and FermiSentence.

It's a good point, and makes me wonder whether we should instead get `map_wires` to work with `FermiWord` and `FermiSentence`. That doesn't need to be within the scope of this PR though! What do you think @soranjh?

Reply:
In this PR, we do not need `map_wires` to work with `FermiWord` and `FermiSentence`.